llvm/lib/Target/RISCV/RISCVISelDAGToDAG.cpp

```diff
@@ -1164,6 +1164,15 @@ void RISCVDAGToDAGISel::Select(SDNode *Node) {
     }
 
     unsigned LShAmt = Subtarget->getXLen() - TrailingOnes;
+    if (Subtarget->hasVendorXTHeadBb() && TrailingOnes > ShAmt) {
```

Comment:
TrailingOnes > ShAmt is always true here, it was checked on line 1142.

Reply:
Thanks! done